testing/mocks/generic.go

```diff
@@ -81,20 +96,114 @@ var (
 		},
 	}
 
+	GenericEventTypes        = []flow.EventType{"deposit", "withdrawal"}
```

Comment:
Since they are generic, I don't see why they would need to be `"deposit"` and `"withdrawal"` for our tests. I would do it like this:

```go
func GenericEventType(index int) flow.EventType {
  return flow.EventType(fmt.Sprintf("generic-event-type-%d", index))
}

func GenericEventTypes(num int) []flow.EventType {
  types := make([]flow.EventType, 0, num)
  for i := 0; i < num; i ++ {
    types = append(types, GenericEventType(i))
  }
  return types
}
```

I would apply this throughout the generic mock types for everything where we want multiple ones. It will deterministically generate the same types given the same number, so behaves the same as an array. At the same time, there is no bound on how many we can use.

Reply:
I didn't see your message before writing my implementations, so I simply used `math/rand` to generate everything instead, with fixed seeds. I think it does the job as well and avoids any bias from our side, WDYT?

Reply:
Everything looks awesome now.